### PR TITLE
[14.0][MIG] membership_prorate: Migration to 14.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa TT32110

Main changes:
- Adapt code to the way invoices are handled in v14.0
- Adapt tests code

ping @chienandalu @pedrobaeza

### CODE REVIEW

File: membership_prorate/models/account_invoice.py [hunk 1]
```diff
@@ -39,22 +40,34 @@ def _prepare_invoice_line_prorate_vals(self, invoice_line):
...
+                quantity = invoice_line_vals["quantity"]
+                invoice = invoice_line.move_id
+                with Form(invoice) as invoice_form:
+                    for index in range(len(invoice.invoice_line_ids)):
+                        if (
+                            invoice_form.invoice_line_ids.edit(index)._values["id"]
+                            == invoice_line.id
+                        ):
+                            with invoice_form.invoice_line_ids.edit(index) as line_form:
+                                line_form.quantity = quantity
```
Comment (chienandalu) [inline membership_prorate/models/account_invoice.py]:
```suggestion
                with Form(invoice) as invoice_form:
                    index = next((
                        index for (index, d) in enumerate(
                            invoice_form.invoice_line_ids._records
                        ) if d["id"] == invoice_line.id), None
                    )
                    if index is not None:
                        with invoice_form.invoice_line_ids.edit(index) as line_form:
                            line_form.quantity = quantity
```

**Review by chienandalu — COMMENTED** [review]
I've been triying to find a more performant way to change the line, but in the end this one looks like the more resilient. A little suggestion to avoid calling the edit() twice:

File: membership_prorate/models/account_invoice.py (hunk 1 above)
Comment (ernestotejeda) [reply membership_prorate/models/account_invoice.py]: Thanks @chienandalu  , nice code by the way :D

**Comment by ernestotejeda** [comment]
Changes done

**Review by chienandalu — APPROVED** [review]
Thanks ;)

**Pull request merged**